packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts

```diff
@@ -0,0 +1,225 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/** Properties for creating a cluster with EksCreateCluster */
+export interface EksCreateClusterProps extends sfn.TaskStateBaseProps {
+  /** Name of the cluster */
+  readonly name: string;
+
+  /**
+   * The desired Kubernetes version for your cluster.
+   * @default - latest
+   */
+  readonly kubernetesVersion?: string;
+
+  /** EKS Role ARN to create a cluster */
+  readonly role: string;
+
+  /** VPC configuration used by the cluster control plane */
+  readonly resourcesVpcConfig: VpcConfigRequest;
+
+  /**
+   * Enable or disable exporting the Kubernetes control plane logs for your cluster to CloudWatch Logs.
+   * @default - no logging configuration
+   */
+  readonly loggingConfiguration?: LoggingConfig;
+
+  /**
+   * Unique, case-sensitive identifier that you provide to ensure the idempotency of the request.
+   * @default - no client request token
+   */
+  readonly clientRequestToken?: string;
+
+  /**
+   * Metadata that you apply to the cluster to assist with categorization and organization
+   * @default - no tags
+   */
+  readonly tags?: { [key: string]: string };
+
+  /**
+   * The encryption configuration for the cluster.
+   * @default - No encryption config
+   */
+  readonly encryptionConfig?: EncryptionConfig[];
```

Comment:
is no encryption a good default for EKS? - i would think that encryption is enabled by default with an ability to turn it off.

Reply:
From what I've seen in usage in the Console and docs here (https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/) under How does it work?, one has to enable encryption.

Reply:
okay, let's point to the documentation about how it needs to be enabled.
feels strange to be able to configure something that cannot be enabled here.